server/portal/apps/site_search/api/views.py

```diff
@@ -0,0 +1,92 @@
+from django.http import JsonResponse
+from elasticsearch_dsl import Search
+from portal.libs.agave.operations import search as search_operation
+from portal.views.base import BaseApiView
+from django.conf import settings
+import logging
+logger = logging.getLogger(__name__)
+
+
+def cms_search(query_string, offset=0, limit=10):
+    cms_index = settings.ES_INDEX_PREFIX.format('cms')
+    cms_search = Search(index=cms_index)\
+        .query(
+        "query_string",
+        query=query_string,
+        default_operator="and",
+        fields=['title', 'body'])\
+        .highlight(
+        'body',
+        fragment_size=100)\
+        .highlight('title')\
+        .highlight_options(
+        pre_tags=["<b>"],
+        post_tags=["</b>"],
+        require_field_match=False)\
+        .extra(from_=offset, size=limit)
+
+    cms_search = cms_search.execute()
+    res = cms_search.hits
+    total = cms_search.hits.total.value
+
+    results = list(map(lambda x: {**x.to_dict(),
+                                  'highlight': x.meta.highlight.to_dict()},
+                       res))
+    return total, results
+
+
+def files_search(query_string, system, offset=0, limit=10):
+    res = search_operation(None, system, '/', offset=offset,
+                           limit=limit, query_string=query_string)
+    return (res['count'], res['listing'])
+
+
+class IndexView(BaseApiView):
+    """
+    Main workbench view.
+    """
+    template_name = 'portal/apps/workbench/index.html'
+
+    def dispatch(self, request, *args, **kwargs):
```

Comment:
is there a reason to use the override  of dispatch?  could this be refactored to use a class-based view or a function-based view?

I am suggesting the refactor just to match things up with other api routes and thinking about the error handling and logging in `BaseApiView::dispatch` which currently circumvented here.

Reply:
No reason, I copied another view and didn't realize it has `dispatch` instead of `get`.